src/nwbinspector/checks/_tables.py

```diff
@@ -292,3 +293,61 @@ def check_table_time_columns_are_not_negative(table: DynamicTable) -> Optional[I
                 )
 
     return None
+
+
+@register_check(importance=Importance.CRITICAL, neurodata_type=TimeIntervals)
+def check_time_intervals_duration(
+    time_intervals: TimeIntervals, duration_threshold: float = MAX_DURATION
+) -> Optional[InspectorMessage]:
+    """
+    Check if the duration spanned by time columns in a TimeIntervals table exceeds a threshold.
+
+    This check examines start_time, stop_time, and any other columns ending in _time.
+
+    Best Practice: :ref:`best_practice_time_interval_time_columns`
+
+    Parameters
+    ----------
+    time_intervals: TimeIntervals
+        The table to check
+    duration_threshold: float, optional
+        Maximum expected duration in seconds. Default is 1 year (365.25 days).
+    """
+    if len(time_intervals.id) == 0:
+        return None
+
+    start_times = []
+    end_times = []
+
+    # Check for start_time and stop_time columns
+    if "start_time" in time_intervals.colnames and len(time_intervals["start_time"]) > 0:
```

Comment:
Here and with the other time columns we are assuming that the time columns are already well-ordered. Is there a way of running checks in order? Maybe this check does not make sense if the other fails and the output will confuse rather than clarify if the ascending order check fails.